ui/src/app/change_password/page.tsx

```diff
@@ -0,0 +1,100 @@
+"use client"
+
+import { changePassword } from "../queries"
+import { useMutation } from "react-query"
+import { useState, ChangeEvent } from "react"
+import { useRouter } from "next/navigation"
+import { useAuth } from "../auth/authContext"
+import { passwordIsValid } from "../utils"
+
+export default function ChangePasswordPage() {
+    const router = useRouter()
+    const auth = useAuth()
+    if (!auth.user) {
+        router.push('/login')
+    }
+
+    const mutation = useMutation(changePassword, {
+        onSuccess: (e) => {
+            setErrorText("")
+            router.push('/certificate_requests')
+        },
+        onError: (e: Error) => {
+            setErrorText(e.message)
+        }
+    })
+
+    const [password1, setPassword1] = useState<string>("")
+    const [password2, setPassword2] = useState<string>("")
+    const [showPassword1, setShowPassword1] = useState<boolean>(false)
+    const passwordsMatch = password1 === password2
+    const [errorText, setErrorText] = useState<string>("")
+    const handlePassword1Change = (event: ChangeEvent<HTMLInputElement>) => { setPassword1(event.target.value) }
+    const handlePassword2Change = (event: ChangeEvent<HTMLInputElement>) => { setPassword2(event.target.value) }
+    return (
+        <div style={{
+            display: "flex",
+            alignContent: "center",
+            justifyContent: "center",
+            flexWrap: "wrap",
+            height: "100vh",
+        }}>
+            <div className="p-panel" style={{
+                width: "45rem",
+                minWidth: "min-content",
+                minHeight: "min-content",
+            }}>
+                <div className="p-panel__content">
+                    <div className="u-fixed-width">
+                        <form className={"p-form-validation " + (!passwordsMatch ? "is-error" : "")}>
+                            <fieldset>
+                                <h4 className="p-panel__title">Change Password</h4>
+                                <label className="p-form__label">New Password</label>
```

Comment:
When the password does not have the requirements (ex. capital letter) it's not possible to submit but there is no red badge to tell us why we did something wrong.

Reply:
I implemented 
> Have the button disabled until the input values become acceptable
from the issue. But if i do this and the above comment, then the form will become red immediately.

A good middle ground could be to only have an error once the user starts inputting text to the second field. I've made the changes, please take a look.

Reply:
I could make it so the errors with the password validation show up first. It would behave very similarly to updating when unfocused.

There isn't really a standard for this, I can find examples for every version. It's just about what we want.

Reply:
I've made some changes, please check again.

Reply:
This feels better in my opinion